src/Json.Schema.ToDotNet/ComparerCodeGenerator.cs

```diff
@@ -0,0 +1,1603 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.Json.Schema.ToDotNet
+{
+    /// <summary>
+    /// Generates classes that implement <see cref="System.Collections.Generic.IComparer{T}"/>.
+    /// </summary>
+    internal class ComparerCodeGenerator
+    {
+        internal const string InstancePropertyName = "Instance";
+
+        private const string UriClassName = "Uri";
+        private const string GenericTypeName = "T";
+        private const string ObjectTypeName = "object";
+        private const string CountPropertyName = "Count";
+        private const string GenericKeyTypeName = "TKey";
+        private const string GenericValueTypeName = "TValue";
+        private const string FirstEqualsArgumentName = "left";
+        private const string SecondEqualsArgumentName = "right";
+        private const string CompareResultVariableName = "compareResult";
+        private const string UriOrignalStringPropertyName = "OriginalString";
+
+        private const string ComparerSuffix = "Comparer";
+        private const string IListInterfaceName = "IList";
+        private const string IComparerInterfaceName = "IComparer";
+        private const string IComparableInterfaceName = "IComparable";
+        private const string IDictionaryInterfaceName = "IDictionary";
+
+        private const string UriComparesMethodName = "UriCompares";
+        private const string ListComparesMethodName = "ListCompares";
+        private const string CompareDelegateName = "compareFunction";
+        private const string ObjectComparesMethodName = "ObjectCompares";
+        private const string ComparerExtensionsClassName = "ComparerExtensions";
+        private const string ListComparesHelperMethodName = "ListComparesHelper";
```

Comment:
>ListComparesHelperMethodName = "ListComparesHelper"; [](http://example.com/codeflow?start=29&length=52)

ListComparesHelper = nameof(ListComparesHelper).

actually not sure if this compiles. :)